### ISSUE TITLE
ablation study on stronger baseline results

### ISSUE THREAD

**Issue by sarah-zhu** [JDAI-CV/fast-reid#56]
Hi, 
This is amazing source code for reid. I've been following the strong baseline results for a long time, and I saw here's stronger baseline results. Do you have ablation study on this stronger baseline? I am curious which one has the major impact.

Thanks ^ ^

**Comment by L1aoXingyu** [comment]
Hi, thanks for your attention to this work.
We will release a paper about this toolkit a week or two, and you can find an ablation study on the stronger baseline.

**Comment by L1aoXingyu** [comment]
@sarah-zhu 
hi, this is the ablation study we conduct experiment on DukeMTMC.
![image](https://user-images.githubusercontent.com/19245504/83222482-fc710900-a1aa-11ea-8582-9d69158bbe7a.png)

**Comment by sarah-zhu** [comment]
@L1aoXingyu 
Thanks! This is really helpful! 
I just tried few tricks: Freeze backbone, Auto-Augment, Cosine LR, Circle Loss. I tested on my own dataset which contains ~3000 identities, 100k images. I use BoT(https://github.com/michuanhaohao/reid-strong-baseline) as my baseline. My observation is:

<img width="580" alt="Screen Shot 2020-05-28 at 11 33 12 PM" src="https://user-images.githubusercontent.com/6743378/83228987-0dcefc80-a13c-11ea-8f99-3974be8c716c.png">

I have few questions:
1. Adding auto-augment will decrease the mAP, which matches with your ablation study. How can I benefit from it?
2. Adding circle loss also doesn't help. I am using both triplet loss and softmax loss. It seems very promising at the beginning of the training (the val mAP gets very high at first 20 epochs, usually I got this number after epoch 60), but then it never goes up. Did you have the same observation? Do you have any suggestions?
3. In your configure file, you use 2000 iterations for warmup and freeze backbone. Should I keep the same amount of iterations if I am using a different dataset? My current configure sets them to 10 epochs. Is it reasonable?

Any suggestions will be helpful. Thank you!

**Comment by L1aoXingyu** [comment]
Hi @sarah-zhu,
1. I found AutoAugment will be helpful in MSMT17, which contains indoor and outdoor scenes, I guess maybe it will helpful for the model's generalization, so I add it.
2. Batch size is important for circle loss. In our experiments, 4 identities and 16 images for each identity in a batch is a good practice.
3. I think warmup iters might be matched with dataset size, 10 epochs are equivalent to 2000 iters in Market and Duke. I got this from [train imagenet in 1 hour](https://arxiv.org/pdf/1706.02677.pdf).

**Comment by sarah-zhu** [comment]
@L1aoXingyu， 谢谢！
确实batch size对于circle loss的影响很大。试了下你说的 4x16的设置，一下子提升了～5%，但是不太理解为什么。

**Comment by L1aoXingyu** [comment]
这个结论我们也是通过实验发现的，猜测可能是每个 batch 里面 id 太多的话，这种 margin-based loss 不好优化。

**Comment by jxxk1986** [comment]
@sarah-zhu ,您好！
请问使用circle loss时，需要triplet loss同时训练吗？如果4*16的PK设置，会不会降低triplet loss的难例挖掘效果？

**Comment by sarah-zhu** [comment]
@jxxk1986 ,我还没有加triplet loss，之前加过效果不好，还要继续调试。

**Comment by sarah-zhu** [comment]
@L1aoXingyu 对于negative pair margin我有点困惑，无论是paper还是大家的实现，都是`s_n - delta_n,` 可以我觉得是`s_n+delta_n`，这样才是給negative pair增加难度。paper里公示（2）第一行也是`s_n+m`, 而且从distance distribution来看，加上circle loss之后，negative pair distance 变小了。下面第一张图是没有circle loss，第二张图是加了circle loss之后。
![distance_his_strongbaseline_merge](https://user-images.githubusercontent.com/6743378/84328035-b9614f00-ab35-11ea-95ac-7355a1f55bce.png)
![distance_his_strongerbaseline_merge](https://user-images.githubusercontent.com/6743378/84328043-be260300-ab35-11ea-942f-97a9156f118f.png)
